src/meadowdb/reader.py

```diff
@@ -240,12 +235,19 @@ def to_pd(self) -> pd.DataFrame:
         # we have to iterate newest partitions first so we know what to filter out of
         # older partitions
         for i, data_file in enumerate(reversed(self._data_list)):
-            if data_file.data_file_type == "write":
+            if (
+                data_file.data_file_type == "write"
+                and data_file.data_filename is not None
```

Comment:
No, only if it's `delete_all`, by construction (see writer code). I only added that line to make mypy happy as it's otherwise redundant by construction as you imply. If that's not there, it complains about `None` in line 247 below.

If you prefer, I can remove this line and just add a type: ignore to those lines though.

I also played with the idea of splitting `DataFileEntry` into three "union cases" `Write filename | Delete filename | DeleteAll` each represented by a little class, so this problem is completely avoided.

Any preference?

Reply:
Ahhhhh thank you, that makes sense. I mean I love the union types, so would definitely vote for that, but don't feel compelled to do it?